lib/enum/arrays.rb

```diff
@@ -1,24 +1,39 @@
+require 'pry-byebug'
+
 class Array
 
-  def sum_of_numeric_elements
+  def initialize (elements)
+    @elements = e
+  end
+
+  def sum_of_numeric_elements #using e for elements.
+    select { [e] e.is_a? Numeric }.reduce(:+) #is_a? to return a boolean value, that if e is a numeric element then add them and return the sum.
   end
 
   def product_of_numeric_elements
+    select { [e] e.is_a? Numeric }.reduce(:*)
   end
 
   def even_numeric_elements
+    select { [e] e.is_a? Numeric && (elem % 2 == 0)}
```

Comment:
Keep your local variable consistent here. If you define it as `|e|`, then you can't call it `elem` later on. Again, this is where running your tests and committing every time you write a new method helps a lot.